bindings/nodejs/lib/types/block/payload/transaction/essence.ts

```diff
@@ -16,56 +19,67 @@ enum TransactionEssenceType {
 }
 
 abstract class TransactionEssence {
-    private type: TransactionEssenceType;
+    readonly type: TransactionEssenceType;
 
     constructor(type: TransactionEssenceType) {
         this.type = type;
     }
-
-    /**
-     * The type of essence.
-     */
-    getType(): TransactionEssenceType {
-        return this.type;
-    }
 }
 
 /**
  * RegularTransactionEssence transaction essence.
  */
 class RegularTransactionEssence extends TransactionEssence {
-    /// The unique value denoting whether the block was meant for mainnet, testnet, or a private network.
-    networkId: number;
-    inputsCommitment: HexEncodedString;
+    /**
+     * The unique value denoting whether the block was meant for mainnet, testnet, or a private network.
+     */
+    networkId: u64;
+
+    @Type(() => Input, {
+        discriminator: ContextInputDiscriminator,
+    })
+    contextInput: ContextInput[];
 
     @Type(() => Input, {
         discriminator: InputDiscriminator,
     })
-    inputs: [Input];
+    inputs: Input[];
+
+    inputsCommitment: HexEncodedString;
 
     @Type(() => Output, {
         discriminator: OutputDiscriminator,
     })
-    outputs: [Output];
+    outputs: Output[];
+
+    allotments: ManaAllotment[];
 
     @Type(() => Payload, {
         discriminator: PayloadDiscriminator,
     })
-    payload: Payload | undefined;
+    payload?: Payload;
+
+    creationSlot?: SlotIndex;
 
     constructor(
-        networkId: number,
+        networkId: u64,
+        contextInput: ContextInput[],
+        inputs: Input[],
         inputsCommitment: HexEncodedString,
-        inputs: [Input],
-        outputs: [Output],
-        payload: Payload | undefined,
+        outputs: Output[],
+        allotments: ManaAllotment[],
```

Comment:
This should be an ordered set. Sets in JS are not ordered, so we have to use arrays, but there's no validation for duplicates, we can validate that in the constructor. If we don't and the array has duplicates, an error would be returned from somewhere in Rust at a later point, not when an instance is created ofc.

Reply:
Since that validation is done on the rust side, it is not needed here.